pkg/logstructured/logstructured.go

```diff
@@ -274,81 +357,65 @@ func (l *LogStructured) ttlEvents(ctx context.Context) chan *server.Event {
 
 	go func() {
 		defer wg.Done()
```

Comment:
Make sure to close the channel. If you don't do this and there is an error in the for loop that returns early, the read from the channel will block forever.
```suggestion
		defer close(lastListRevision)
		defer wg.Done()
```